modern/src/map/MapPositions.js

```diff
@@ -162,34 +172,41 @@ const MapPositions = ({ positions, onClick, showStatus, selectedPosition, titleF
       map.off('click', clusters, onClusterClick);
       map.off('click', onMapClick);
 
-      if (map.getLayer(id)) {
-        map.removeLayer(id);
-      }
       if (map.getLayer(clusters)) {
         map.removeLayer(clusters);
       }
-      if (map.getLayer(direction)) {
-        map.removeLayer(direction);
-      }
-      if (map.getSource(id)) {
-        map.removeSource(id);
-      }
+
+      [id, selected].forEach((id) => {
```

Comment:
Actually the `id` in the lambda is very confusing. It clashes with the component id.